### PR TITLE
Kkraune/test doc

### PR DESCRIPTION
**Pull request by kkraune**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

cleanup, remove redundancies

### CODE REVIEW

File: documentation/testing.html [hunk 1]
```diff
@@ -208,3 +184,28 @@ <h3 id="using-configuration-builders">Using configuration builders</h3>
...
+<p>
+A System Test is a test that tests a full Vespa application, using its endpoints.
+A System Test is invoked before a change to the production environment:
+<ol>
+  <li>Application package deployment</li>
+  <li>Vespa software upgrade</li>
+</ol>
+Vespa System Tests use <a href="https://junit.org/junit5/">JUnit 5</a> annotations.
+Tests are annotated with <code>@Test</code> as unit tests.
+The test class must be annotated with <code>@Tag("SystemTest")</code>
```
Comment (jonmv) [inline documentation/testing.html]: Needs to be `ai.vespa.hosted.cd.SystemTest`, which is held in the static `ai.vespa.tenant-cd.SystemTest.name` field.

File: documentation/testing.html [hunk 2]
```diff
@@ -3,59 +3,35 @@
...
-Notice the <code>com.yahoo.application.container.JDisc</code> class that is accessed by
-the test through <code>app.getJDisc(clusterName)</code>. That class has methods for
-exercising all common <a href="component-types.html">component types</a>.
+The <code><a href="http://javadoc.io/page/com.yahoo.vespa/application/latest/com/yahoo/application/Application.html">
+Application</a></code> class is useful when writing unit tests.
+Application uses the application package configuration and set up a container instance for testing.
+The <code><a href="http://javadoc.io/page/com.yahoo.vespa/application/latest/com/yahoo/application/container/JDisc.html">
+JDisc</a></code> class that is accessed by
+the test through <code><a href="https://javadoc.io/page/com.yahoo.vespa/application/latest/com/yahoo/application/Application.html#getJDisc-java.lang.String-">app.getJDisc(clusterName)</a></code> -
+this class has methods for exercising all common <a href="component-types.html">component types</a>.
```
Comment (jonmv) [inline documentation/testing.html]: This sentence isn't very good English :)

File: documentation/testing.html (hunk 1 above)
Comment (jonmv) [reply documentation/testing.html]: This can be changed later, though.

File: documentation/testing.html [hunk 3]
```diff
@@ -208,3 +184,28 @@ <h3 id="using-configuration-builders">Using configuration builders</h3>
...
+<ol>
+  <li>Application package deployment</li>
+  <li>Vespa software upgrade</li>
+</ol>
+Vespa System Tests use <a href="https://junit.org/junit5/">JUnit 5</a> annotations.
+Tests are annotated with <code>@Test</code> as unit tests.
+The test class must be annotated with <code>@Tag("SystemTest")</code>
+to be a System Test class (and hence invoked in correct phase).
+See <code>test.categories</code> in the example
+<a href="https://github.com/vespa-engine/sample-apps/blob/master/basic-search-java/pom.xml">
```
Comment (jonmv) [inline documentation/testing.html]: This must point to the implementation we own, in `tenant-base/pom.xml`.

File: documentation/testing.html [hunk 4]
```diff
@@ -208,3 +184,28 @@ <h3 id="using-configuration-builders">Using configuration builders</h3>
...
+pom.xml</a> for details. To run a system test:
+<ol>
+  <li>Deploy the application, make note of endpoints</li>
+  <li>Update <code>test-config</code> ToDo: update filename</li>
+  <li>Run tests using <pre>$ mvn test -Dtest.categories=SystemTest</pre></li>
```
Comment (jonmv) [inline documentation/testing.html]: Likewise, this should be a profile from the parent pom (in `tenant-base`).

File: documentation/testing.html (hunk 1 above)
Comment (kkraune) [reply documentation/testing.html]: ok, but why ai.vespa.hosted.cd.SystemTest?  this is just instrumentation to maven for which tests to run - or is there more to it?

Comment (jonmv) [reply documentation/testing.html]: Right now that's the required name (comes from the @Category, which needs a class argument). We can change it, or allow both. If it doesn't need to work today, and I guess it doesn't, then "SystemTest" is fine.

**Comment by kkraune** [comment]
yes we put this out now that is not wrong for self-hosted - keeping this in one place

**Pull request merged**